src/tests/ftest/util/ior_test_base.py

```diff
@@ -0,0 +1,151 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import os
+
+from apricot import TestWithServers
+from daos_api import DaosPool
+from ior_utils import IorCommand, IorFailed
+from mpio_utils import MpioUtils
+
+
+class IorTestBase(TestWithServers):
+    """Base IOR test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a IorTestBase object."""
+        super(IorTestBase, self).__init__(*args, **kwargs)
+        self.ior_cmd = None
+        self.processes = None
+        self.hostfile_clients_slots = None
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super(IorTestBase, self).setUp()
+
+        # Get the parameters for IOR
+        self.ior_cmd = IorCommand()
+        self.ior_cmd.set_params(self)
+        self.processes = self.params.get("np", '/run/ior/client_processes/*')
+
+    def tearDown(self):
+        """Tear down each test case."""
+        try:
+            if self.pool is not None and self.pool.attached:
+                self.pool.destroy(1)
+        finally:
+            # Stop the servers and agents
+            super(IorTestBase, self).tearDown()
+
+    def run_ior_with_pool(self):
+        """Execute ior with optional overrides for ior flags and object_class.
+
+        If specified the ior flags and ior daos object class parameters will
+        override the values read from the yaml file.
+
+        Args:
+            ior_flags (str, optional): ior flags. Defaults to None.
+            object_class (str, optional): daos object class. Defaults to None.
+        """
+        # Get the parameters used to create a pool
+        mode = self.params.get("mode", "/run/pool/*")
+        uid = os.geteuid()
+        gid = os.getegid()
+        group = self.params.get("setname", "/run/pool/*", self.server_group)
+        scm_size = self.params.get("scm_size", "/run/pool/*")
+        nvme_size = self.params.get("nvme_size", "/run/pool/*", 0)
+        svcn = self.params.get("svcn", "/run/pool/*", 1)
+
+        # Initialize a python pool object then create the underlying
+        # daos storage
+        self.pool = DaosPool(self.context)
+        self.pool.create(
+            mode, uid, gid, scm_size, group, None, None, svcn, nvme_size)
```

Comment:
I'm on the fence with that one.  Although there are a few tests that use a single pool, I think it makes sense to have an easier to use class for testing 1 or more pools - hence the `general_utils.py` `TestPool` class.

I also think it makes sense to consider trying to limit the setUp() and tearDown() methods to the server/agent code as it could make the tests easier to port to an environment where those were always running - if we will ever need that.